File: Tutorial5-3Dagger-Android-MutipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_mutipleactivitiesandfragments/main/MainActivityModule.java
package com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.main;

import android.content.Context;

import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.di.ActivityScope;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.main.MainActivity;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.model.ToastMaker;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class MainActivityModule {

    /*
    NOTE: This can only be MainActivity(AndroidInjector<T>), Activity does not work
     */
    @Binds
    abstract Context bindContext(MainActivity activity);

    @ActivityScope
    @Provides
    public static ToastMaker provideToastMaker(Context context) {
        return new ToastMaker(context);
    }
}

File: Tutorial6-4ComponentFactoryDagger-Android-AAC/src/main/java/com/smarttoolfactory/tutorial6_4componentfactorydagger_android/di/module/MainActivityModule.kt
package com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.module

import android.content.Context
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.scope.ActivityScope
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.main.MainActivity
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.model.DummyDependency
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
interface MainActivityModule {

    companion object {

        // any bindings here are effectively static and part of the module
        @ActivityScope
        @JvmStatic
        @Provides
        fun provideDummyDependency(context: Context): DummyDependency {
            return DummyDependency(context)
        }
    }

    /*
    NOTE: This can only be MainActivity(AndroidInjector<T>), Activity does not work
     */
    @Binds
    abstract fun bindContext(activity: MainActivity): Context

}
